Do not run NativeImageBuildContainerRunnerTest test unless -Dtest-containers is set

holly-cummins opened the pull request:
Fixes https://github.com/quarkusio/quarkus/issues/25108. 

This is a slightly more elaborate fix than I did for https://github.com/quarkusio/quarkus/issues/25141. Only one test in the `core/deployment` module needs containers, so I didn't want to disable the whole module unless `start-containers` was set. I could have just let the test read the `start-containers` system property, but that would result in it being disabled by default if run within an IDE. I figure if you're running a test within an IDE you want it to run, without any extra configuration, so I inverted the properties in the pom so the test is disabled by default when run in maven, but enabled by default when run 'loose'. 

With the changes, on a docker/podman-free system, 
- Running in an IDE runs the test and gives a failure 
- `./mvnw -Dquickly -DskipTests=false -Dstart-containers -f core/deployment` runs the test and gives a failure 
- `./mvnw -Dquickly -DskipTests=false -f core/deployment` runs all the other tests in the module, and runs clean

holly-cummins committed:
Do not run NativeImageBuildContainerRunnerTest test unless -Dtest-containers is set.

geoand commented on core/deployment/src/test/java/io/quarkus/deployment/pkg/steps/NativeImageBuildContainerRunnerTest.java:
```diff
@@ -14,6 +15,9 @@ class NativeImageBuildContainerRunnerTest {
 
     @Test
     void testBuilderImageBeingPickedUp() {
+        // This will default to false in the maven build and true in the IDE, so this will still run if invoked explicitly
+        Assumptions.assumeFalse(Boolean.parseBoolean(System.getProperty("avoid-containers")),
+                "Skipping native image build container tests because -Dstart-containers is not set.");
```
Can't we replace this change with 

```
@EnabledIfSystemProperty(named="avoid-containers", matches="true")
```

?

holly-cummins replied:
Yes, that would be cleaner and more consistent with the rest of the codebase, thanks! It would have to be `@DisabledIfSystemProperty(named="avoid-containers", matches="true")` because of the inversion I did. 
(... which is slightly less consistent with the rest of the codebase, and more complicated, but it felt more desirable.)

geoand replied:
Good point!

holly-cummins commented:
Updated and squashed. I've tested locally and confirmed the behaviour is still the same (with no podman, fail in IDE, fail with `mvn -DskipTests=false -Dstart-containers`, pass with just `mvn -DskipTests=false`.

holly-cummins commented:
I've just spotted another instance of this, so I'll apply the same fix to that module and resquash.

geoand reviewed (approved):
Thanks!

famod commented on core/deployment/pom.xml:
```diff
@@ -133,6 +133,18 @@
+            <plugin>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <systemProperties>
```
This is deprecated: https://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#systemProperties

> Deprecated. Use systemPropertyVariables instead.

geoand replied:
Ah, good catch!

geoand replied:
Probably best to just change this now and we can deal with the others in another PR
